library/systemd/src/lib/yast2/system_service.rb

```diff
@@ -0,0 +1,490 @@
+# find current contact information at www.suse.com.
+
+require "forwardable"
+
+Yast.import "SystemdService"
+
+module Yast2
+  # This class represents a system service
+  #
+  # When talking about systemd, it might happen that a service is compose by a set of units
+  # (services, sockets, paths and so on). This class is able to group those units and
+  # offer an API to handle them together.
+  #
+  # Additionally, this class does not modify the underlying system until the {#save}
+  # method is called.
+  #
+  # @example Enabling a service
+  #   cups = SystemService.find("cups")
+  #   cups.start_mode = :on_boot
+  #   cups.save
+  #
+  # @example Activating a service
+  #   cups = SystemService.find("cups")
+  #   cups.start
+  #   cups.save
+  #
+  # @example Ignoring status changes on 1st stage
+  #   cups = SystemService.find("cups")
+  #   cups.start
+  #   cups.save(keep_state: true)
+  class SystemService
+    extend Forwardable
+
+    # Error when a service is not found
+    class NotFoundError < RuntimeError; end
+
+    # @return [Yast::SystemdServiceClass::Service]
+    attr_reader :service
+
+    # @return [Hash<Symbol,Object>] Errors when trying to write changes to the
+    #   underlying system.
+    attr_reader :errors
+
+    # @return [Symbol, nil] :start, :stop, :restart or :reload. It returns nil
+    #   if no action has been requested yet.
+    attr_reader :action
+
+    # @!method support_reload?
+    #
+    # @return [Boolean]
+    def_delegator :@service, :can_reload?, :support_reload?
+
+    def_delegators :@service, :name, :static?, :running?, :description
+
+    class << self
+      # Finds a service by its name
+      #
+      # @param name [String] service name
+      # @return [SystemService, nil] nil if the service is not found
+      def find(name)
+        systemd_service = Yast::SystemdService.find(name)
+        return nil unless systemd_service
+
+        new(systemd_service)
+      end
+
+      # Finds a service by its name
+      #
+      # @param name [String] service name
+      #
+      # @raise [NotFoundError] if the service is not found
+      # @return [SystemService]
+      def find!(name)
+        system_service = find(name)
+        raise NotFoundError unless system_service
+
+        system_service
+      end
+
+      # Finds a set of services by their names
+      #
+      # @param names [Array<String>] service names to find
+      #
+      # @raise [NotFoundError] if any service is not found
+      # @return [Array<SystemService>]
+      def find_many(names)
+        systemd_services = Yast::SystemdService.find_many(names)
+        raise NotFoundError if systemd_services.any?(&:nil?)
+
+        systemd_services.map { |s| new(s) }
+      end
+    end
+
+    # Constructor
+    #
+    # @param service [Yast::SystemdServiceClass::Service]
+    def initialize(service)
+      @service = service
+      @changes = {}
+      @errors = {}
+    end
+
+    # State of the service
```

Comment:
"state" is a very overloaded term. Why not name this method `active_state` ? That would also kind of document the allowed values of the string.

Reply:
Well, we have discarded `#active_state` because we want the API to be as agnostic as possible. `#active_state` is too related to systemd.